ksqldb-engine/src/test/java/io/confluent/ksql/query/QueryRegistryImplTest.java

```diff
@@ -483,6 +499,18 @@ public void shouldCallSandboxOnClose() {
     verify(listener2, times(0)).onClose(any());
   }
 
+  @Test
+  public void shouldReplaceQueryfromOldRuntimeUsingOldRuntime() {
+    sharedRuntimes = false;
+    QueryMetadata query = givenCreate(registry, "q1", "source",
+        Optional.of("sink1"), CREATE_AS);
+    assertThat("does not use old runtime", query instanceof PersistentQueryMetadataImpl);
+    sharedRuntimes = true;
+    query = givenCreate(registry, "q1", "source",
+        Optional.of("sink1"), CREATE_AS);
+    assertThat("does not use old runtime", query instanceof PersistentQueryMetadataImpl);
```

Comment:
Is it worth adding to this test case what happens when adding a new query?  

E.g., once sharing is enabled, does the old query go to the old runtime because all queries would go there?

Reply:
It shouldn't be necessary but I don't see any harm in adding it.

Reply:
Thanks!